cfme/infrastructure/host.py

```diff
@@ -452,7 +448,12 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+        from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+        dest = 'Hosts'
+        if hasattr(self.obj, 'provider') and self.obj.provider:
```

Comment:
@mfalesni well, I agree with your suggestion. In this case, by the way, most changes from this PR are excess. So we can easily revert it.
@mshriver are you okay with it? I remember we were talking about 'asking for permission' variant at first